libraries/libraries_api.go

```diff
@@ -83,7 +83,7 @@ func (a LibrariesAPI) WaitForLibrariesInstalled(wait Wait) (result *ClusterLibra
 			return resource.RetryableError(err)
 		}
 		if err != nil {
-			return resource.NonRetryableError(err)
+			return resource.NonRetryableError(wrapClusterStatusAPIError(err, wait.ClusterID))
```

Comment:
okay, so please simplify the logic here between lines 81 and 87. the new logic should be something like: if we get error from `a.ClusterStatus`, return it as non-retriable.

you probably need to directly paste only these lines:

```go
        apiErr, ok := err.(common.APIError)
	if !ok {
		return resource.NonRetryableError(err)
	}
	// fix non-compliant error code
	if apiErr.StatusCode != 404 && strings.Contains(apiErr.Message, 
                fmt.Sprintf("Cluster %s does not exist", id)) {
		apiErr.StatusCode = 404
	}
        return resource.NonRetryableError(apiErr)
```

then amount of unit tests will be smaller ;)

Reply:
@nfx Updated per your suggestion, please review and confirm if we are good.